storage/data-plane/src/test/java/com/microsoft/azure/storage/APISpec.groovy

```diff
@@ -5,26 +5,40 @@ package com.microsoft.azure.storage
 
 import com.microsoft.aad.adal4j.AuthenticationContext
 import com.microsoft.aad.adal4j.ClientCredential
+import com.microsoft.azure.management.resources.core.TestBase
 import com.microsoft.azure.storage.blob.*
 import com.microsoft.azure.storage.blob.models.*
+import com.microsoft.azure.storage.interceptor.InterceptorManager
+import com.microsoft.azure.storage.interceptor.TestResourceNamer
 import com.microsoft.rest.v2.Context
 import com.microsoft.rest.v2.http.*
+import com.microsoft.rest.v2.policy.DecodingPolicyFactory
 import com.microsoft.rest.v2.policy.RequestPolicy
 import com.microsoft.rest.v2.policy.RequestPolicyFactory
+import com.microsoft.rest.v2.util.FlowableUtil
 import io.reactivex.Flowable
 import io.reactivex.Single
-import org.junit.Assume
+import org.junit.Rule
+import org.junit.rules.TestName
 import org.spockframework.lang.ISpecificationContext
 import spock.lang.Shared
 import spock.lang.Specification
 
-import java.lang.reflect.Method
+import java.lang.reflect.Field
 import java.nio.ByteBuffer
+import java.nio.channels.AsynchronousFileChannel
 import java.time.OffsetDateTime
 import java.util.concurrent.Executors
 
 class APISpec extends Specification {
-    static final String RECORD_MODE = "RECORD"
+    static final String RECORD_MODE = "RECORDING"
```

Comment:
why RECORDING?
AZURE_TEST_MODE variable is globally set to RECORD for all SDKs in the live pipeline.

Reply:
The record mode should different than live mode. Our pipeline set to RECORD for live mode. I have to differentiate the actual record mode, as the record mode will write things to json.

Reply:
RECORDING: record mode
LIVE or RECORD: live mode
PLAYBACK: playback mode

Reply:
Can you add a comment block that includes the values so future developers can know what to set? Or an enum type

Reply:
I have added comments here.

Reply:
These terms are confusing.
Ideally, RECORD and PLAYBACK should only exist.
Record will write things to Json under target folder, will not overwrite the recordings we have checked in to github (src/test/resources/session-records)
Don't see much value in making logging to file optional through an extra type.

Reply:
The storage track 1.5 has some tests which record files larger than 2GB, so both JS and Java needs to differentiate these tests and let large file tests only run in live mode without recording.

Reply:
Storage track 2 will address this issue as you said we only leave record and playback modes only.